tests/frontend/test_circuitspecs_chip0.py

```diff
@@ -0,0 +1,300 @@
+            ops.Interferometer(U) | (q[2], q[3])
+            ops.MeasureFock() | q
+
+        res = prog.compile("chip0")
+
+        prog = sf.Program(4)
+
+        with prog.context as q:
+            ops.S2gate(0.5, 0) | (q[0], q[2])
+            ops.S2gate(0.5, 0) | (q[1], q[3])
+            ops.Interferometer(U, mesh="rectangular_symmetric", drop_identity=False) | (q[0], q[1])
+            ops.Interferometer(U, mesh="rectangular_symmetric", drop_identity=False) | (q[2], q[3])
+            ops.MeasureFock() | q
+
+        expected = prog.compile(DummyCircuit())
+
+        for cmd1, cmd2 in zip(res.circuit, expected.circuit):
+            assert cmd1.op.__class__ == cmd2.op.__class__
+            assert [p.x for p in cmd1.op.p] == [p.x for p in cmd2.op.p]
+            assert [i.ind for i in cmd1.reg] == [i.ind for i in cmd2.reg]
+
+    def test_unitaries_do_not_match(self):
+        """Test exception raised if the unitary applied to modes [0, 1] is
+        different to the unitary applied to modes [2, 3]"""
+        prog = sf.Program(4)
+        U = random_interferometer(2)
+
+        with prog.context as q:
+            ops.S2gate(0.5) | (q[0], q[2])
+            ops.S2gate(0.5) | (q[1], q[3])
+            ops.Interferometer(U) | (q[0], q[1])
+            ops.Interferometer(U) | (q[2], q[3])
+            ops.BSgate() | (q[2], q[3])
+            ops.MeasureFock() | q
+
+        with pytest.raises(CircuitError, match="must be identical to interferometer"):
+            res = prog.compile("chip0")
+
+    def test_unitary_too_large(self):
+        """Test exception raised if the unitary is applied to more
+        than just modes [0, 1] and [2, 3]."""
+        prog = sf.Program(4)
+        U = random_interferometer(4)
+
+        with prog.context as q:
+            ops.S2gate(0.5) | (q[0], q[2])
+            ops.S2gate(0.5) | (q[1], q[3])
+            ops.Interferometer(U) | q
+            ops.BSgate() | (q[2], q[3])
+            ops.MeasureFock() | q
+
+        with pytest.raises(CircuitError, match="must be applied separately"):
+            res = prog.compile("chip0")
+
+    def test_mach_zehnder_interferometers(self):
+        """Test Mach-Zehnder gates correctly compile"""
+        prog = sf.Program(4)
+        phi = 0.543
+        theta = -1.654
+
+        with prog.context as q:
+            ops.S2gate(0.5) | (q[0], q[2])
+            ops.S2gate(0.5) | (q[1], q[3])
+            ops.MZgate(phi, theta) | (q[0], q[1])
+            ops.MZgate(phi, theta) | (q[2], q[3])
+            ops.MeasureFock() | q
+
+        res = prog.compile("chip0")
+
+        expected = sf.Program(4)
+
+        with expected.context as q:
+            ops.S2gate(0.5, 0) | (q[0], q[2])
+            ops.S2gate(0.5, 0) | (q[1], q[3])
+
+            # corresponds to MZgate(phi, theta) on modes [0, 1]
+            ops.Rgate(phi) | q[0]
+            ops.BSgate(np.pi/4, np.pi/2) | (q[0], q[1])
+            ops.Rgate(theta+2*np.pi) | q[0]

```

Comment:
Yep, the issue is the `sf.decompositions.rectangular_decomposition` function. The best approach is to modify it so that the output decomposition parameters are always `phi % 2*np.pi`, or something similar, to force it within the range `[0, 2*pi)`.

@lneuhaus, what range of angle parameters do you prefer on your end, to minimize the number of conversions we have to do?

Reply:
`[0, 2*pi)` is the best for me.

Reply:
Fixed, now `Interferometer(U, mesh="rectangular_symmetric")` now always enforces `phi % (2*np.pi)` on all phase angles.